lib/brain.js

```diff
@@ -2985,7 +2991,15 @@ Brain.prototype._submitPromoCode = function _submitPromoCode (data) {
   const promoCode = data.input
 
   this.trader.verifyPromoCode(promoCode, this.tx).then(res => {
-    this.tx = Tx.update(this.tx, { discount: res.promoCode.discount })
+    /*
+      * It's possible for a customer to insert a promo code before inserting the phone data, leading to two possible discounts.
+      * In that case, use the bigger discount.
+      * See: verifyCustomerDiscounts()
+      */
+    const currentDiscount = this.tx.discount || 0
+    if (res.promoCode.discount > currentDiscount) {
+      this.tx = Tx.update(this.tx, { discount: res.promoCode.discount, discountSource: 'promoCode' })
```

Comment:
Instead of having a discount source it might be better to just pass down a flag whenever we use a coupon code. We just update the tx with something like "promoCodeApplied".

If we do that we'll be able to replace this part of the code with the regular `verifyCustomerDiscounts` and fix the bug mentioned on my other comment.

Reply:
Not sure if I got the comment correctly (as that bug in specifically does not occur). But I found another interaction during cash-out where this change comes in handy. I'll work on this